resource_estimator/src/estimates/error_correction/code_with_threshold_and_distance.rs

```diff
@@ -115,19 +129,25 @@ impl<Evaluator: CodeWithThresholdAndDistanceEvaluator> ErrorCorrection
         qubit: &Self::Qubit,
         required_logical_qubit_error_rate: f64,
     ) -> Result<u64, String> {
-        let physical_error_rate = self.evaluator.physical_error_rate(qubit);
-        let numerator = 2.0 * (self.crossing_prefactor / required_logical_qubit_error_rate).ln();
-        let denominator = (self.error_correction_threshold / physical_error_rate).ln();
-
-        let code_distance = (((numerator / denominator) - 1.0).ceil() as u64) | 0x1;
-
-        if let Some(max_distance) = self.max_code_distance {
-            if max_distance < code_distance {
-                return Err(format!("The computed code distance {code_distance} is too high; maximum allowed code distance is {max_distance}; try increasing the total logical error budget"));
+        // if distance coefficient power is 0, the function can be inverted and we can avoid looping over all code parameters
+        if self.distance_coefficient_power == 0 {
```

Comment:
Suggest moving the function code itself (above) and the inverse function code to the separate Rust functions to make it easier to understand.

Reply:
Thanks, resolved in recent commit.